src/middlewares.ts

```diff
@@ -16,13 +16,19 @@ const hashPattern = /(?:^|.*?\.)hash-([a-f0-9]+)\./;
 function assembleSubdomainUrlForHash(req: any, commitHash: CommitHash) {
   const protocol = req.secure || req.headers.host.indexOf( 'calypso.live' ) > -1 ? "https" : "http";
 
+  const query = 
+    Object.keys( req.query )
+    .reduce( ( q, key ) => key === 'branch' ? q : q.concat( `${ key }=${ q[ key as keyof {} ] }` ), [] )
```

Comment:
Express seems to decode the query:

https://expressjs.com/en/api.html#req.query

Should we passing key and value through `encodeURIComponent` to build the URL below?

Reply:
sounds like a good catch there @sirreal and yah I agree - let's encode them. I'm AFK now for a couple weeks so feel free to commandeer the PR.